internal/protocol/connection_id.go

```diff
@@ -17,7 +19,38 @@ func GenerateConnectionID() (ConnectionID, error) {
 	return ConnectionID(b), nil
 }
 
+// ReadConnectionID reads a connection ID of length len from the given io.Reader.
+// It returns io.EOF if there are not enough bytes to read.
+func ReadConnectionID(r io.Reader, len int) (ConnectionID, error) {
+	if len == 0 {
+		return nil, nil
+	}
+	c := make(ConnectionID, len)
+	_, err := io.ReadFull(r, c)
+	if err == io.ErrUnexpectedEOF {
+		return nil, io.EOF
+	}
+	return c, err
+}
+
 // Equal says if two connection IDs are equal
 func (c ConnectionID) Equal(other ConnectionID) bool {
 	return bytes.Equal(c, other)
```

Comment:
Actually, this would only be relevant for the server, if the attacker tries to guess the connection ID. But the compares there are performed by the map, and we don't have any influence on that anyway.